apps/admin-dashboard/app/routes/_dashboard.events.$id.add.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+  type ActionFunctionArgs,
+  json,
+  type LoaderFunctionArgs,
+  redirect,
+} from '@remix-run/node';
+import {
+  Form as RemixForm,
+  useActionData,
+  useNavigate,
+} from '@remix-run/react';
+import { addLink } from 'app/shared/queries/events';
+
+import { Event } from '@oyster/types';
+import {
+  Button,
+  Form,
+  getActionErrors,
+  Input,
+  Modal,
+  validateForm,
+} from '@oyster/ui';
+
+import { Route } from '../shared/constants';
+import {
+  commitSession,
+  ensureUserAuthenticated,
+  toast,
+} from '../shared/session.server';
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  await ensureUserAuthenticated(request);
+
+  return json({});
+}
+
+const CreateEventFormData = Event.pick({
+  link: true,
+});
+
+export async function action({ params, request }: ActionFunctionArgs) {
+  const session = await ensureUserAuthenticated(request);
+
+  const form = await request.formData();
+
+  const { data, errors } = validateForm(
+    CreateEventFormData,
+    Object.fromEntries(form)
+  );
+
+  if (!data) {
+    return json({
+      error: 'Something went wrong, please try again to upload event link.',
+      errors,
+    });
+  }
+
+  await addLink(params.id as string, data.link as string);
+
+  toast(session, {
+    message: `Link uploaded successfully.`,
+    type: 'success',
+  });
+
+  return redirect(Route.EVENTS, {
+    headers: {
+      'Set-Cookie': await commitSession(session),
+    },
+  });
+}
+
+export default function AddEventLink() {
+  const navigate = useNavigate();
+
+  function onClose() {
+    navigate(Route.EVENTS);
+  }
+
+  return (
+    <Modal onClose={onClose}>
+      <Modal.Header>
+        <Modal.Title>Add Event URL</Modal.Title>
+        <Modal.CloseButton />
+      </Modal.Header>
+      <Modal.Description>Please Add a link for the event</Modal.Description>
+
+      <AddEventLinkForm />
+    </Modal>
+  );
+}
+
+const { link } = CreateEventFormData.keyof().enum;
+
+function AddEventLinkForm() {
+  const { error, errors } = getActionErrors(useActionData<typeof action>());
+
+  return (
+    <RemixForm className="form" method="post">
+      <Form.Field error={errors.link} label="Link" labelFor={link} required>
+        <Input id={link} name={link} required />
```

Comment:
If the event already has an event recording, I think it'd be clearer that the value would override the existing value if we set the `defaultValue` to the existing link. Can we add that?

You can use the [`getEvent`](https://github.com/colorstackorg/oyster/blob/5d3b522e87614bf03b95a268dcefef95ae44905b/packages/core/src/modules/event/queries/get-event.ts#L8) method in the `loader`!

Reply:
Made all other changes. This particular one could still use some work. Is there an option for a placeholder value on form? That may be better than what I currently have.

<img width="666" alt="Screenshot 2024-04-08 at 11 00 26 PM" src="https://github.com/colorstackorg/oyster/assets/121850312/a83c261f-9c15-46e7-9026-5ff637283bd4">

Reply:
So actually HTML inputs have this attribute called [`defaultValue`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement#defaultvalue) which will automatically load the value into the input when the form is loaded! Meaning if you don't change that input, it'll still retain that value.

Wherever possible, we'll try to use the web standards and this is a good example of that! HTML FTW!